test/e2e/taints_test.go

```diff
@@ -0,0 +1,335 @@
+/*
```

Comment:
I know it's a lot of work but can you add a test with three pods on the same node
* one tolerates T without tolerationSeconds
* one tolerates T for some tolerationSeconds==t
* one does not tolerate T
Then add taint T to the node and verify that the first one is never evicted, second is evicted after t, and third is evicted immediately?

A couple of other ideas for tests
* same pod configuration as above, but add two taints T and S, and verify all three pods are evicted immediately
* a pod that tolerates two taints, with different tolerationSeconds, then add both taints and verify that the pod is evicted after the min of the tolerationSeconds